File: server/Models/Album.cs
using System.ComponentModel.DataAnnotations;
namespace post_it_dotnet_fullstack.Models;

public class Album : RepoItem<int>
{
  [MinLength(3), MaxLength(25)] public string Title { get; set; }
  [MinLength(15), MaxLength(250)] public string Description { get; set; }
  [Url, MaxLength(2000)] public string CoverImg { get; set; }
  public bool Archived { get; set; }
  public string Category { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
}

File: client/src/models/Album.js
import { Profile } from "./Account.js"

export class Album {
  constructor(data) {
    this.id = data.id
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
    this.title = data.title
    this.coverImg = data.coverImg
    this.archived = data.archived
    this.description = data.description
    this.category = data.category
    this.creatorId = data.creatorId
    this.creator = new Profile(data.creator)
  }
}